services/web/server/tests/unit/with_postgres/test_storage.py

```diff
@@ -108,6 +135,48 @@ def storage_server(loop, aiohttp_server, app_cfg, aiohttp_unused_port):
         assert len(data) == 1
         assert data[0]['user_id'] == logged_user['id']
 
+@pytest.mark.parametrize("role,expected", [
+    (UserRole.ANONYMOUS, web.HTTPUnauthorized),
+    (UserRole.GUEST, web.HTTPForbidden),
+    (UserRole.USER, web.HTTPOk),
+    (UserRole.TESTER, web.HTTPOk),
+])
+async def test_get_datasets_metadata(client, storage_server, logged_user, role, expected):
+    url = "/v0/storage/locations/0/datasets"
```

Comment:
MINOR: try using more readable variables e.g. ``expected_url`` and ``got_url``  vs ``url`` and ``_url`` 
That leads to more readable code as ``assert expected_url == got_url`` (both ``URL``)